### PR TITLE
gadget: tweaks to DiskStructureDeviceTraits + expand test cases

### PR DESCRIPTION
**Pull request by anonymouse64**
After working on DOS disks again I realized we left out a couple settings, and that we don't need to include the ID, since it is a bit ambiguous and difficult to compare between two hypothetical devices.

Also add a DOS unit test case from a real Raspi.

### CODE REVIEW

**Review by mardy — APPROVED** [review]
Reviewed it commit by commit, LGTM!

File: gadget/gadget.go [hunk 1]
```diff
@@ -264,18 +264,20 @@ type DiskStructureDeviceTraits struct {
 	OriginalKernelPath string `json:"kernel-path"`
 	// PartitionUUID is the partuuid as defined by i.e. /dev/disk/by-partuuid
 	PartitionUUID string `json:"partition-uuid"`
-	// FilesystemLabel is the label of the filesystem for structures that have
-	// filesystems, i.e. /dev/disk/by-label
-	FilesystemLabel string `json:"filesystem-label"`
 	// PartitionLabel is the label of the partition for GPT disks, i.e.
 	// /dev/disk/by-partlabel
 	PartitionLabel string `json:"partition-label"`
+	// PartitionType is the type of the partition i.e. 0x83 for ext4 on DOS, or
```
Comment (bboozzoo) [inline gadget/gadget.go:270]: 0x83 does not necessarily imply ext4, but means a Linux filesystem in general, usually was ext2,3,4 but could also be other filesystems (reiserfs, xfs, and so on). AFAIU it's the same with GPT type.

**Review by bboozzoo — APPROVED** [review]

File: gadget/gadget.go (hunk 1 above)
Comment (mvo5) [reply gadget/gadget.go]: I tweaked this comment based on your feedback to unblock this. Will merge once it's green :)

Comment (anonymouse64) [reply gadget/gadget.go]: you beat me to it @mvo5 :smile: thanks for this change

Comment (mvo5) [reply gadget/gadget.go]: I really hope I did not duplicate work but took some work from your shoulders. If so happy to help if not my apologies.

Comment (anonymouse64) [reply gadget/gadget.go]: no worries just a single comment I hadn't committed yet

**Pull request merged**